galsim/wfirst/wfirst_wcs.py

```diff
@@ -0,0 +1,571 @@
+
+        for i in range(n_sip):
+            for j in range(n_sip):
+                if i+j >= 2 and i+j < n_sip:
+                    sipstr = "A_%d_%d"%(i,j)
+                    header[sipstr] = a_sip[i_sca,i,j]
+                    sipstr = "B_%d_%d"%(i,j)
+                    header[sipstr] = b_sip[i_sca,i,j]
+
+        if not as_header:
+            wcs = galsim.FitsWCS(header=header)
+            wcs_list.append(wcs)
+        else:
+            wcs_list.append(header)
+
+    return wcs_list
+
+def findSCA(wcs_list, ra=None, dec=None, pos=None, include_border=False):
+    """
+    This is a subroutine to take a list of WCS (one per SCA) from galsim.wfirst.getWCS() and query
+    which SCA a particular real-world coordinate would be located on.  If the position is not
+    located on any of the SCAs, the result will be None.
+
+    The position should be specified in a way similar to how it is specified for
+    galsim.wfirst.getWCS().
+
+    Depending on what the user wants to do with the results, they may wish to use the
+    `include_border` keyword.  This keyword determines whether or not to include an additional
+    border corresponding to half of the gaps between SCAs.  For example, if a user is drawing a
+    single image they may wish to only know whether a given position falls onto an SCA, and if so,
+    which one (ignoring everything in the gaps).  In contrast, a user who plans to make a sequence
+    of dithered images might find it most useful to know whether the position is either on an SCA or
+    close enough that in a typical dither sequence it might appear on the SCA at some point.  Use of
+    `include_border` switches between these scenarios.
+
+    @param ra               Right ascension of the sky position of interest, as a galsim.Angle.
+                            Must be provided with `dec`.
+    @param dec              Declination of the sky position of interest, as a galsim.Angle.  Must be
+                            provided with `ra`.
+    @param pos              A galsim.CelestialCoord indicating the sky position of interest, as an
+                            alternative to providing `ra` and `dec`.
+    @param include_border   If True, then include the half-border around SCA to cover the gap
+                            between each sensor. [default: False]
+    @returns an integer value of the SCA on which the position falls, or None if the position is not
+             on any SCA.
+
+    """
+    if len(wcs_list) != galsim.wfirst.n_sca+1:
+        raise ValueError("wcs_list should be a list of length %d output by"
+                         " galsim.wfirst.getWCS!"%(galsim.wfirst.n_sca+1))
+
+    # Parse input position
+    use_ra, use_dec = _parse_input_position(ra, dec, pos)
+
+    # Set up the minimum and maximum pixel values, depending on whether or not to include the
+    # border.
+    min_x_pix, max_x_pix, min_y_pix, max_y_pix = _calculate_minmax_pix(include_border)
```

Comment:
Use `BoundsI` here for the bounding box.  And then the `includes` method is what you want below to see if `image_pos` is inside the bounds.